src/Controller/Component/QueryParamPreserverComponent.php

```diff
@@ -85,11 +86,10 @@ public function apply()
         $request = $this->getController()->request;
         $key = $this->_hashKey();
 
-        if (empty($request->getQuery()) && $request->getSession()->check($key)) {
```

Comment:
Why did you revert the 3.6 changes?

Reply:
No idea! It's been too long for me to remember.